### ISSUE TITLE
crashed deck app

### ISSUE THREAD

**Issue by petrisch** [stefan-niedermann/nextcloud-deck#938]
Crash maybe already reported, i have no idea :-)

```
Full Crash:

App Version: 1.17.4
App Version Code: 1017004
App Flavor: fdroid

Files App Version Code: 30150190

---

OS Version: 3.18.120-perf(00WW_6_19F)
OS API Level: 28
Device: PLE
Manufacturer: HMD Global
Model (and Product): TA-1033 (TA-1033_00WW)

---

java.lang.RuntimeException: Exception while computing database live data.
	at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:92)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: Activity_tmp.type (code 1299 SQLITE_CONSTRAINT_NOTNULL)
	at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
	at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:748)
	at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:754)
	at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:64)
	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1771)
	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1698)
	at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.execSQL(FrameworkSQLiteDatabase.java:242)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.migration.Migration_30_31.migrate(Migration_30_31.java:31)
	at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:99)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:177)
	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:398)
	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:298)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:145)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:106)
	at androidx.room.RoomDatabase.inTransaction(RoomDatabase.java:476)
	at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.java:281)
	at androidx.room.RoomDatabase.query(RoomDatabase.java:324)
	at androidx.room.util.DBUtil.query(DBUtil.java:83)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao.AccountDao_Impl$5.call(AccountDao_Impl.java:276)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao.AccountDao_Impl$5.call(AccountDao_Impl.java:273)
	at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:90)
	... 3 more

```

**Comment by petrisch** [comment]
Clearing app data and reconecting helps, so no real issue. Aand great new feature is there, thanks a lot!!!!

**Comment by stefan-niedermann** [comment]
Will be fixed in `1.17.7`, just wait for it to be published by F-Droid.

If you are sure that you don't have unsynchronized data, you can also [clear your the storage](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/FAQ.md#i-have-experienced-an-error) to fix the issue.
